### DISCUSSION TITLE
Table component

### DISCUSSION THREAD

**Discussion by KrisLau** [react-native-elements/react-native-elements#3794] [RFC]
**Is your feature request related to a problem? Please Describe.**
A lot of people resort to external libraries to create tables but there's really not a lot of options out there and the most popular one just got [deprecated](https://github.com/dohooo/react-native-table-component). It would be nice to be able to not have to get a separate library just for a table. 

Personally I'm struggling to recreate something like this:
![image](https://github.com/react-native-elements/react-native-elements/assets/46275463/d05d8c90-ab41-4814-bf1d-2c92443c110f)
where the players column does not move when the user scrolls horizontally through the columns and the header row is also locked

**Describe the solution you'd like**
I think something like [`react-native-datatable-component`](https://github.com/MuhammadRafeh/react-native-datatable-component) with more customizability (ability to change icons & icon position) or maybe add features like filtering, locking some columns and rows so that those are pinned / don't scroll. Would also be extremely helpful to have horizontal scrolling for the columns if they are more than the width of the screen ([current workaround](https://github.com/MuhammadRafeh/react-native-datatable-component/issues/16#issuecomment-1563402969) isnt terrible though).

Also I think it would be good to remove the prop `colNames` and `colSettings` as they are redundant. It should be replaced by the prop name `columns` and the input should be the same as `colSettings`.

Example of implementation:
```js
import React from 'react';
import DataTable, { COL_TYPES } from 'react-native-datatable-component';

const SomeComponent = props => {
    return (
        <DataTable
            data={[ 
                { name: 'Muhammad Rafeh', age: 21, gender: 'male' },
                { name: 'Muhammad Akif', age: 22, gender: 'male' },
                { name: 'Muhammad Umar', age: 21, gender: 'male' },
                { name: 'Amna Shakeel', age: 22, gender: 'female' },
                { name: 'Muhammad Ammar', age: 20, gender: 'male' },
                { name: 'Muhammad Moiz', age: 13, gender: 'male' }
            ]}
            // original props that should be removed
            /* colNames={['name', 'age', 'gender']}
            colSettings={[
              { name: 'name', type: COL_TYPES.STRING, width: '40%' }, 
              { name: 'age', type: COL_TYPES.INT, width: '30%' }, 
              {name: 'gender', type: COL_TYPES.STRING, width: '30%'}
            ]}
            pages={2} */
           columns={[
              { name: 'name', type: COL_TYPES.STRING, width: '40%' }, 
              { name: 'age', type: COL_TYPES.INT, width: '30%' }, 
              {name: 'gender', type: COL_TYPES.STRING, width: '30%'}
            ]}
            rowsPerPage={10}
            backgroundColor={'rgba(23,2,4,0.2)'} 
            headerLabelStyle={{ color: 'grey', fontSize: 12 }} 
        />
    );
}

export default SomeComponent;
```
`doSort` should also be renamed to just `sort` I think and the sorting icon should indicate if it is sorted by ascending or descending order

**Describe alternatives you've considered**
Those two and [`react-native-reanimated-table`](https://github.com/dohooo/react-native-reanimated-table) are the only ones I've seen

-  I feel like the dohoo repos use a weird structure for the input data (arrays for each row of data).  
- `react-native-datatable-component` uses JSON which seems more intuitive imo but it's not very popular and only has one dev. 
- React Native Paper has a [Data Table](https://callstack.github.io/react-native-paper/3.0/data-table.html) component but it doesn't really help with rendering a table as it feels just like a pre-styled row and you would still need to manually map each row of your data. There's no way to pass a whole array of data.

**Additional context**
Maybe the author of `react-native-datatable-component` could be contacted to merge his repo into RNE and then add to the existing code from there to add filtering and allow for more customization.

**Comment by KrisLau** [comment]
Made a PR to `react-native-datatable-component` to address some of those issues & add some of those features which would probably help with implementing this

**Reply by arpitBhalla** [reply]
Hey, @KrisLau, Let me know if you are willing to work on it.
